Stripe/STPAddressViewModel.m

```diff
@@ -48,71 +48,108 @@ - (instancetype)initWithRequiredBillingFields:(STPBillingAddressFields)requiredB
+                                             [[STPAddressFieldTableViewCell alloc] initWithType:STPAddressFieldTypeState contents:@"" lastInList:NO delegate:self],
+                                             // Postal code cell will be added later if necessary
+                                             [[STPAddressFieldTableViewCell alloc] initWithType:STPAddressFieldTypeCountry contents:_addressFieldTableViewCountryCode lastInList:NO delegate:self],
+                                             ] mutableCopy];
+#if __IPHONE_OS_VERSION_MAX_ALLOWED >= 80300
+            if (requiredShippingAddressFields & PKAddressFieldName) {
+                [postalCells removeObjectAtIndex:0];
+            }
+#endif
+            [cells addObjectsFromArray:postalCells];
+        }
+        if (requiredShippingAddressFields & PKAddressFieldPhone) {
+            [cells addObject:[[STPAddressFieldTableViewCell alloc] initWithType:STPAddressFieldTypePhone contents:@"" lastInList:NO delegate:self]];
+        }
+        STPAddressFieldTableViewCell *lastCell = [cells lastObject];
+        if (lastCell != nil) {
+            lastCell.lastInList = YES;
+        }
+        _addressCells = [cells copy];
+        [self commonInit];
+    }
+    return self;
+}
+
+- (void)commonInit {
+    _addressFieldTableViewCountryCode = [[NSLocale autoupdatingCurrentLocale] objectForKey:NSLocaleCountryCode];
+    [self updatePostalCodeCellIfNecessary];
+}
+
 - (void)updatePostalCodeCellIfNecessary {
     STPPostalCodeType postalCodeType = [STPPostalCodeValidator postalCodeTypeForCountryCode:_addressFieldTableViewCountryCode];
     BOOL shouldBeShowingPostalCode = (postalCodeType != STPCountryPostalCodeTypeNotRequired);
     if (shouldBeShowingPostalCode && !self.showingPostalCodeCell) {
-        switch (self.requiredBillingAddressFields) {
-            case STPBillingAddressFieldsNone:
-                // Do nothing
-                break;
-            case STPBillingAddressFieldsZip:
-                self.addressCells = @[
-                                      [[STPAddressFieldTableViewCell alloc] initWithType:STPAddressFieldTypeZip contents:@"" lastInList:YES delegate:self]
-                                      ];
-                [self.delegate addressViewModel:self addedCellAtIndex:0];
+        if (self.isBillingAddress && self.requiredBillingAddressFields == STPBillingAddressFieldsZip) {
+            self.addressCells = @[
+                                  [[STPAddressFieldTableViewCell alloc] initWithType:STPAddressFieldTypeZip contents:@"" lastInList:YES delegate:self]
+                                  ];
+            [self.delegate addressViewModel:self addedCellAtIndex:0];
+            [self.delegate addressViewModelDidChange:self];
+        }
+        else if ((self.isBillingAddress && self.requiredBillingAddressFields == STPBillingAddressFieldsFull) ||
+                 (!self.isBillingAddress && (self.requiredShippingAddressFields & PKAddressFieldPostalAddress))) {
```

Comment:
This check should be `((self.requiredShippingAddressFields & PKAddressFieldPostalAddress) == PKAddressFieldPostalAddress)` (See http://sam.dods.co/blog/2014/02/01/checking-for-a-value-in-a-bit-mask/ )

Also since these `if` statement checks are getting a bit long confusing, it might be worth separating out the logic into some other method or enum. Especially since they have to exactly match the logic in the opposite branch below.